independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/options/BootstrapMavenOptions.java

```diff
@@ -41,12 +44,18 @@
     public static final String CHECKSUM_WARNING_POLICY = "c";
     public static final String BATCH_MODE = "B";
     public static final String NO_TRANSFER_PROGRESS = "ntp";
+    public static final String USER_PROP_PREFIX = "-D";
```

Comment:
This constant shouldn't be exposed, it should be private.
The constants above are actually matching CLIManager ones. I'd probably keep it this way. There is CLIManager.SET_SYSTEM_PROPERTY which we could add here and also add support for querying BootstrapMavenOptions for the properties.

Reply:
Not sure I follow here… I also believe that adding such feature is outside the scope of this PR anyway.